File: old_backend/important-core-code/go/vulnerable_sql.go
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical SQL injection vulnerabilities for testing Aikido scanner

package main

import (
    "database/sql"
    "fmt"
    "net/http"
    _ "github.com/lib/pq"
)

func getUserHandler(w http.ResponseWriter, r *http.Request) {
    userId := r.URL.Query().Get("id")
    
    db, _ := sql.Open("postgres", "user=admin password=secret123 dbname=mydb")
    defer db.Close()
    
    // VULNERABLE: String concatenation in query
    query := "SELECT * FROM users WHERE id = '" + userId + "'"
    rows, _ := db.Query(query)
    defer rows.Close()
}

func searchHandler(w http.ResponseWriter, r *http.Request) {
    search := r.URL.Query().Get("term")
    
    db, _ := sql.Open("mysql", "root:password@/database")
    
    // VULNERABLE: fmt.Sprintf in SQL
    query := fmt.Sprintf("SELECT * FROM products WHERE name LIKE '%%%s%%'", search)
    db.Query(query)
}

func main() {
    http.HandleFunc("/user", getUserHandler)
    http.HandleFunc("/search", searchHandler)
    http.ListenAndServe(":8080", nil)
}

File: old_backend/important-core-code/javascript/vulnerable-sql.js
// INTENTIONALLY VULNERABLE CODE - FOR SECURITY SCANNER TESTING ONLY
// This file contains critical SQL injection vulnerabilities for testing Aikido scanner

const express = require('express');
const mysql = require('mysql');
const app = express();

const connection = mysql.createConnection({
  host: 'localhost',
  user: 'root',
  password: 'password123',
  database: 'userdb'
});

app.get('/user', (req, res) => {
  const userId = req.query.id;
  // VULNERABLE: String concatenation allows SQL injection
  const query = "SELECT * FROM users WHERE id = '" + userId + "'";
  connection.query(query, (error, results) => {
    res.json(results);
  });
});

app.get('/search', (req, res) => {
  const searchTerm = req.query.term;
  // VULNERABLE: Template literals still vulnerable
  const query = `SELECT * FROM products WHERE name LIKE '%${searchTerm}%'`;
  connection.query(query, (error, results) => {
    res.json(results);
  });
});

app.listen(3000);
